### PR TITLE
AGR-2124 changes to allow the exclusion of faucets

### PR DESCRIPTION
**Pull request by adamgibs**
Frontend changes for AGR-2124

### CODE REVIEW

File: src/lib/searchHelpers.js [hunk 1]
```diff
@@ -181,6 +203,30 @@ export function parseQueryString(queryString) {
   });
 }
 
+//adds a '-' to the beginning of a query parameter to indicate that it should be excluded
+export function markAsExcluded(queryObject, value){
+  let qp = JSON.parse(JSON.stringify(queryObject)) || {};
```
Comment (paaatrick) [inline src/lib/searchHelpers.js:208]: Is this here for deep cloning `queryObject`? Elsewhere in this file the lodash `clone` function is used for that purpose. Maybe that could be used here as well?

**Review by paaatrick — COMMENTED** [review]
A few small comments on the changes, and it looks like there are some ESLint warnings that need to get cleaned up. Get in touch with me if you're not sure how to find and/or fix them.

This seems to all function correctly, but overall all the exclude icons end up feeling a bit clutter-y to me. Maybe it can be brought up with the UI group to see if there are other ideas.

File: src/containers/search/filterSelector/singleFilterSelector.js [hunk 2]
```diff
@@ -59,11 +72,14 @@ class SingleFilterSelector extends Component {
     let values = <ul className={style.filterList}>{value.values.map (v => this.renderValue(v))}</ul>;
     return (
       <li className='nav-item' key={_key}>
-        <Link className={`nav-link${classSuffix}`} to={{ pathname: SEARCH_PATH, search: stringifyQuery(newQueryObj) }}>
-          <span className={style.aggLink}>
-            <span className={style.aggLinkLabel}>{nameNode}</span><span>{value.total.toLocaleString()}</span>
-          </span>
-        </Link>
+        <span className={style.aggLinkLine}>
```
[the remark this answers is not in the document]
Comment (adamgibs) [reply src/containers/search/filterSelector/singleFilterSelector.js:75]: Will do.

File: src/lib/searchHelpers.js [hunk 3]
```diff
@@ -23,6 +30,17 @@ export function makeValueDisplayName(unformattedName) {
...
 
+export function isExcluded(value) {
+  if(value){
+    return value.charAt(0) === '-';
+
+  }
+}
+
+export function removeExclude(value){
+  return value = value.substring(1);
```
[the remark this answers is not in the document]
Comment (adamgibs) [reply src/lib/searchHelpers.js:41]: I think I had this in two lines originally, but neglected to remove the assignment after refactoring. Will fix.

File: src/lib/searchHelpers.js [hunk 4]
```diff
@@ -110,6 +128,10 @@ export function makeTitleCaseFieldDisplayName(unformattedName) {
   return toTitleCase(makeFieldDisplayName(unformattedName));
 }
 
+export function addNotToFieldDisplayName(fieldName) {
+  return fieldName = 'NOT ' + fieldName;
```
[the remark this answers is not in the document]
Comment (adamgibs) [reply src/lib/searchHelpers.js:132]: Same as above

File: src/lib/searchHelpers.js (hunk 1 above)
Comment (adamgibs) [reply src/lib/searchHelpers.js:208]: Yep, this is for deep cloning. I should have left a comment indicating that.

**Comment by adamgibs** [comment]
Hey, thanks for taking the time review my code! I'm very bad at UI design so other ideas about how the exclude functionality should look would be much appreciated.

I think the Eslint warnings are from the Windows link breaks (CRLF in Windows as apposed to LF in Unix systems). I'll see what I can do about that.

**Pull request merged**